packages/hw-app-ada/test/src/api/getWalletPublicKeyWithIndex.js

```diff
@@ -0,0 +1,115 @@
+    ada = await getAda();
+  });
+
+  afterEach(async () => {
+    await ada.t.close();
+  });
+
+  it("Should successfully get public key", async () => {
+    const index = 0xFFFFFFFF;
+
+    promptUser("Please accept public key request");
+    
+    const response = await ada.getWalletPublicKeyWithIndex(index);
+    validate(response, schema);
+
+    const publicKey = Buffer.from(response.publicKey, "hex");
+    expect(publicKey.length).to.equal(32);
+  });
+  
+  it("Should return same public key with same index consistently", async () => {
+    const index = 0xFEEDBEEF;
+  
+    promptUser("Please accept public key request");
+
+    const response = await ada.getWalletPublicKeyWithIndex(index);
+    validate(response, schema);
+      
+    promptUser("Please accept public key request");
+
+    let res = await ada.getWalletPublicKeyWithIndex(index);
+    expect(res.publicKey).to.equal(response.publicKey);
+
+    promptUser("Please accept public key request");
+
+    res = await ada.getWalletPublicKeyWithIndex(index);
+    expect(res.publicKey).to.equal(response.publicKey);
+  });
+
+  it("Should successfully get public key for lowest hardened index (0x80000000)", async () => {
+    const index = 0x80000000;
+
+    promptUser("Please accept public key request");
+
+    const response = await ada.getWalletPublicKeyWithIndex(index);
+    validate(response, schema);
+  });
+
+  it("Should not get first non-hardened index (0x7FFFFFFF)", async () => {
+    const index = 0x7FFFFFFF;
+
+    promptUser("Please accept public key request");
+
+    try {
+      const response = await ada.getWalletPublicKeyWithIndex(index);
+      throw new Error("Expected error");
+    } catch (error) {
+      expect(error.message).to.have.string("5201");
+    }
+  });
+
+  it("Should reject invalid index", async () => {
+    try {
+      const response = await ada.getWalletPublicKeyWithIndex("44\"1815\"");
+      throw new Error("Expected error");
+    } catch (error) {
+      expect(error.message).to.have.string("5003");
+    }
+  });
+
+  ifHeadlessIt("Should get public key 20 times (stress test)", (done) => {
+    const index = 0xBABADADA;
+
+    const check = (response) => {
+      validate(response, schema);
+      return ada.getWalletPublicKeyWithIndex(index);
+    };
+
+    ada.getWalletPublicKeyWithIndex(index)
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
```

Comment:
maybe you can use `async` function too & do

```
for (let i=0; i<20; i++) {
  check(await ada.getWalletPublicKeyWithIndex(index))
}
```

?

Reply:
Ah yes, good catch, they're left over from pre `async/await`. There's a few of those so I'll grab them all while I'm at it.